### PR TITLE
fix(hil-serl): drain queue on get_last_item_from_queue

### LINKED ISSUE #1523: [HIL SERL] multiprocessing Queue.get_nowait is not getting all the items in the queue
### System Info

```Shell
- `lerobot` version: 0.1.0
- Platform: Linux-6.11.0-29-generic-x86_64-with-glibc2.39
- Python version: 3.10.13
- Huggingface_hub version: 0.30.2
- Dataset version: 3.5.1
- Numpy version: 2.2.6
- PyTorch version (GPU?): 2.6.0+cu124 (True)
- Cuda version: 12040
- Using GPU in script?: 
`python -m lerobot.scripts.rl.learner --config_path ...`
`python -m lerobot.scripts.rl.actor --config_path ...`
```

### Information

- [ ] One of the scripts in the examples/ folder of LeRobot
- [ ] My own task or dataset (give details below)

### Reproduction

Sometimes the `parameters_queue` on the actor keep growing, and can consume all the RAM.
_It can also potentially means that the actor is not getting the latest policy(did not reviewed this)._

Here can be seen that the "Policy(actor) parameters queue size" keeps growing. The queue size on the learner (Optimization prefix) looks stable.
Image

Similar issue was reported for cases when the queues are used with big items in https://bugs.python.org/issue23582.

I updated the code to drain the queue in `get_last_item_from_queue`, and the memory usage was reduced and become  more stable. The red line is without fix, other lines with the fix code bellow.

Image

```
    # Drain queue and keep only the most recent parameters
    while queue.qsize():
        try:
            while True:
                item = queue.get_nowait()
        except Empty:
            pass
```

### Expected behavior

The parameters_queue should be drain complete to get the latest policy and to avoid OOM errors.
I will open a PR soon

cc @AdilZouitine

### PR DESCRIPTION
**Pull request by jpizarrom**
## What this does
This PR drain the queues.

Examples:
|  Title               | Label           |
|----------------------|-----------------|
| Fixes #1523       | (🐛 Bug)        |

## How it was tested
Explain/show how you tested your changes.

Examples:
- Added `test_something` in `tests/test_stuff.py`.
- Added `new_feature` and checked that training converges with policy X on dataset/environment Y.
- Optimized `some_function`, it now runs X times faster than previously.

## How to checkout & try? (for the reviewer)
Provide a simple way for the reviewer to try out your changes.

Examples:
```bash
pytest -sx tests/test_stuff.py::test_something
```
```bash
python -m lerobot.scripts.train --some.option=true
```

## SECTION TO REMOVE BEFORE SUBMITTING YOUR PR
**Note**: Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR. Try to avoid tagging more than 3 people.

**Note**: Before submitting this PR, please read the [contributor guideline](https://github.com/huggingface/lerobot/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr).

### CODE REVIEW

File: src/lerobot/utils/queue.py [hunk 1]
```diff
@@ -29,11 +30,22 @@ def get_last_item_from_queue(queue: Queue, block=True, timeout: float = 0.1) ->
...
+        except Empty:
+            pass
+    else:
+        # Drain queue and keep only the most recent parameters
+        while queue.qsize():
+            try:
+                while True:
+                    item = queue.get_nowait()
+            except Empty:
+                pass
```
Comment (helper2424) [inline src/lerobot/utils/queue.py]:
```suggestion
    # Drain queue and keep only the most recent parameters
    if platform.system() == "Darwin":
        # On Mac, avoid using `qsize` due to unreliable implementation.
        # There is a comment on `qsize` code in the Python source:
        # Raises NotImplementedError on Mac OSX because of broken sem_getvalue()
        try:
            while True:
                item = queue.get_nowait()
        except Empty:
            pass
            
        return item
    
    while queue.qsize() > 0:
        with suppress(Empty): item = queue.get_nowait()
```

Wdyt about such approach? The code looks cleaner, it will also required to import `from contextlib import suppress
` in to of the file.

Comment (helper2424) [reply src/lerobot/utils/queue.py]: Also, to be sure that it works - please run tests locally, and probably we will need one/two tests with `from torch.multiprocessing import Event, Queue` queue in tests, for now it uses just Queue.

Tests are located there `tests/utils/test_queue.py`

Comment (helper2424) [reply src/lerobot/utils/queue.py]: Also, would be great to add in comments to the method the issue, which u posted into the discord - so people will be in context of that when checking the code.

File: tests/utils/test_queue.py [hunk 2]
```diff
@@ -18,12 +18,16 @@
...
 from queue import Queue
 
+import pytest
+from torch.multiprocessing import Queue as TorchMPQueue
+
 from lerobot.utils.queue import get_last_item_from_queue
 
 
-def test_get_last_item_single_item():
+@pytest.mark.parametrize("queue_cls", [Queue, TorchMPQueue])
```
Comment (jpizarrom) [inline tests/utils/test_queue.py]: @helper2424 i parametrized queue_cls in all tests
wdyt

File: src/lerobot/utils/queue.py (hunk 1 above)
Comment (jpizarrom) [reply src/lerobot/utils/queue.py]: the PR was updated, as per your comments. thanks

File: src/lerobot/utils/queue.py [hunk 3]
```diff
@@ -30,10 +32,18 @@ def get_last_item_from_queue(queue: Queue, block=True, timeout: float = 0.1) ->
         item = None
 
     # Drain queue and keep only the most recent parameters
-    try:
-        while True:
+    if platform.system() == "Darwin":
+        # On Mac, avoid using `qsize` due to unreliable implementation.
+        # There is a comment on `qsize` code in the Python source:
+        # Raises NotImplementedError on Mac OSX because of broken sem_getvalue()
+        with suppress(Empty):
```
Comment (helper2424) [inline src/lerobot/utils/queue.py]: I think that here still need to go with old way in this case, we need loop and conditions to leave the loop:
```
try:
    while True:
        item = queue.get_nowait()
     except:
         pass
```
Because exception for us is a trigger to leave the loop.

File: tests/utils/test_queue.py (hunk 2 above)
Comment (helper2424) [reply tests/utils/test_queue.py]: Good direction, I would do several fixes:
1 - I would don't touch old tests. Before I tried to use `torch.multiprocessing. Queue ` for all tests, and it was working not very well in CI, because torch Queus is very complex - it creates separate threads and processes which cobnflicts during running CI byt some reason.
2 - I would just add two (or maybe just one test), the common one - like add items, and get last one and check that it's correct and that queue is empty. This test will explicitly use `torch.multiprocessing.Queue` instead of normal Queue. 
3 - I would remove `TorchMPQueue` - we don't need it here :)

**Comment by jpizarrom** [comment]
for debugging a send the optimization step to the actor, the actor is taking the latest sent policy. left learner, right actor.

image

File: src/lerobot/utils/queue.py (hunk 3 above)
Comment (jpizarrom) [reply src/lerobot/utils/queue.py]: yes, you are right, fixed

File: tests/utils/test_queue.py (hunk 2 above)
Comment (jpizarrom) [reply tests/utils/test_queue.py]: done
